src/libs/actions/Report.js

```diff
@@ -328,15 +331,24 @@ function fetchChatReportsByIDs(chatList) {
  *
  * @param {Number} reportID
  * @param {Number} sequenceNumber
+ * @param {Boolean} saveNewMarkerPosition
  */
-function setLocalLastRead(reportID, sequenceNumber) {
+function setLocalLastRead(reportID, sequenceNumber, saveNewMarkerPosition) {
     lastReadSequenceNumbers[reportID] = sequenceNumber;
 
     // Update the report optimistically
-    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
-        unreadActionCount: 0,
-        lastVisitedTimestamp: Date.now(),
-    });
+    if (saveNewMarkerPosition) {
+        Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
+            unreadActionCount: Math.max(reportMaxSequenceNumbers[reportID] - sequenceNumber, 0),
+            lastVisitedTimestamp: Date.now(),
+            newMarkerPosition: saveNewMarkerPosition ? sequenceNumber : 0,
```

Comment:
I think that `newMarkerPosition` is a bad name because the word "position" can mean and imply different things. I would suggest naming this something more explicit like `newMarkerSequenceNumber`.